maldoca/ole/oss_utils.cc

```diff
@@ -62,100 +73,89 @@ void CloseUConverter(UConverter** conv) {
-#endif  // _WIN32
   internal_converter_ = InternalConverter::kNone;
   // Fixing missing encoding;
   // cp10000 is calld MAC in iconv
   if (strcmp(encode_name, "cp10000") == 0) {
     encode_name = "MAC";
     DLOG(INFO) << "Replaced cp10000 with MAC";
   }
-#if defined(_WIN32)
-  UErrorCode err_to_unicode;
-  UErrorCode err_to_utf8;
-  converter_to_unicode_ = ucnv_open(encode_name, &err_to_unicode);
-  converter_to_utf8_ = ucnv_open("UTF-8", &err_to_utf8);
-
-  // Both "ucnv_open" have to succeed, otherwise it is considered a fail.
-  if (U_FAILURE(err_to_unicode) || U_FAILURE(err_to_utf8)) {
-    if (U_FAILURE(err_to_unicode)) {
-      HandleUConverterError(converter_to_unicode_, encode_name, err_to_utf8);
-    }
-    if (U_FAILURE(err_to_utf8)) {
-      HandleUConverterError(converter_to_utf8_, "UTF-8", err_to_utf8);
-    }
-#else
+
   converter_ = iconv_open("UTF-8", encode_name);
   if (converter_ == reinterpret_cast<iconv_t>(-1)) {
     converter_ = nullptr;
-    LOG(ERROR) << "Fail to open iconv for '" << encode_name
+    LOG(ERROR) << "Failed to open iconv for '" << encode_name
                << "', error code: " << errno;
-#endif  // _WIN32
     // Windows encoding, we really want to make sure this works so we'll use our
     // own
-#if defined(_WIN32)
-    if (_stricmp(encode_name, "cp1251") == 0) {
-#else
     if (strcasecmp(encode_name, "cp1251") == 0) {
-#endif  // _WIN32
       internal_converter_ = InternalConverter::kCp1251;
       DLOG(INFO) << "Use internal cp1251 encoder";
       return true;
     }
-#if defined(_WIN32)
-    if (_stricmp(encode_name, "cp1252") == 0) {
-#else
     if (strcasecmp(encode_name, "cp1252") == 0) {
-#endif  // _WIN32
       internal_converter_ = InternalConverter::kCp1252;
       DLOG(INFO) << "Use internal cp1252 encoder";
       return true;
     }
-#if defined(_WIN32)
-    if (_stricmp(encode_name, "LATIN1") == 0) {
-#else
     if (strcasecmp(encode_name, "LATIN1") == 0) {
-#endif  // _WIN32
       internal_converter_ = InternalConverter::kLatin1;
       DLOG(INFO) << "Use internal LATIN1 encoder";
       return true;
     }
     return false;
   }
   return true;
+
+#else   // _WIN32
+  // Supported Windows code pages have to be mapped manually.
+  // TODO: add more supported code pages:
+  // https://docs.microsoft.com/en-us/windows/win32/intl/code-page-identifiers
+  if (_stricmp(encode_name, "cp1251") == 0) {
+    code_page_ = 1251;
+  } else if (_stricmp(encode_name, "cp1252") == 0) {
+    code_page_ = 1252;
+  } else if (_stricmp(encode_name, "utf-16le") == 0) {
+    code_page_ = kUtf16LECodePage;
+  } else if (_stricmp(encode_name, "latin1") == 0) {
+    code_page_ = 28591;
+  } else if (_stricmp(encode_name, "cp936") == 0) {
+    code_page_ = 936;
+  } else {
+    init_success_ = false;
+    LOG(ERROR) << "Windows code page is not supported: " << encode_name;
+    return false;
+  }
+  init_success_ = true;
+  DLOG(INFO) << "Using Windows code page: " << code_page_;
+  return true;
```

Comment:
I am surprised windows doesn't have a translation API. In that case I suggest we lowercase encode_name and create a hash_map for the mapping so it's easier to maintain. so you do something like

static const absl::flat_hash_map<std::string, int> * name_to_code_map  = [] () {
  auto* map = new  absl::flat_hash_map<std::string, int> ();
  *map = {{"cp1251", 1251}, ... };
   return map;
};
 auto iter = name_to_code_map->find(tolower(encode_name)); 
...

Reply:
added todo